crates/oxc_linter/src/rules/unicorn/consistent_function_scoping.rs

```diff
@@ -160,6 +160,10 @@ impl Rule for ConsistentFunctionScoping {
     fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
         let (function_declaration_symbol_id, function_body, reporter_span) = match node.kind() {
             AstKind::Function(function) => {
+                if let Some(AstKind::AssignmentExpression(_)) = ctx.nodes().parent_kind(node.id()) {
```

Comment:
I don't know if I agree with this.

This case would now not be reported:
```ts
function outer() {
  const inner = function inner() {}
}
```

IMO, we should be checking for assignment expressions and, when found, using that node's scope id instead of t he function's scope id.

Reply:
Are you sure? I am getting the error.
```
  ⚠ eslint-plugin-unicorn(consistent-function-scoping): Function does not capture any variables from the outer scope.
   ╭─[consistent_function_scoping.tsx:2:31]
 1 │ function outer() {
 2 │                 const inner = function inner() {}
   ·                               ────────
 3 │             }
   ╰────
  help: Move this function to the outer scope.
```

Reply:
ah @DonIsaac this is because that's a variable declaration, whereas this is fixing an assignment expressin.

pesky JS 🙂 

Regardless. i think we should add this as a `fail` test case. 
We should also add

```ts
function outer() { 
    inner = function foo() {}
}
```

as a `fail` test case

Reply:
@camc314 the other thing is if you pay close attention to the underlined keyword of the error, it's the ```function``` keyword which is underlined. The correct keyword should be the ```inner``` token.

Why should we add the second example you mentioned as a failed test? It is getting passed in Eslint.